libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioAdapter.cs

```diff
@@ -269,7 +269,27 @@ private Activity ReadRequest(HttpRequest httpRequest)
                 Text = twilioEvent.Body,
                 ChannelData = twilioEvent,
                 Type = ActivityTypes.Message,
+                Attachments = GetMessageAttachments(twilioEvent),
             };
         }
+
+        private List<Attachment> GetMessageAttachments(TwilioEvent message)
+        {
+            var attachments = new List<Attachment>();
+            if (int.TryParse(message.NumMedia, out var numMediaResult) && numMediaResult > 0)
+            {
+                for (var i = 0; i < numMediaResult; i++)
```

Comment:
Could this be a for each loop? Didn't ReSharper suggest it?

Reply:
How would you do that? And how would you translate the index-access below in MediaContentTypes and MediaUrls?

Reply:
I guess it can't be done because _MediaUrls_ and _MediaContentTypes_ are two different lists.